### PR TITLE
ARROW-7166: [Java] Remove redundant code for Jdbc adapters

### PR DESCRIPTION
**Pull request by liyafan82**
As discussed in https://github.com/apache/arrow/pull/5508#issuecomment-543011016, we need a separate issue to extract common logic to a common super class.

This makes the code clearer, and we need to make sure we have no performance regression.

### CODE REVIEW

**Comment by liyafan82** [comment]
Benchmark evaluations show no performance degradation:

Before:
JdbcAdapterBenchmarks.consumeBenchmark      avgt    5   547.267 ±   8.121  us/op
JdbcAdapterBenchmarks.consumeRowsBenchmark  avgt    5   593.117 ± 109.100  us/op
JdbcAdapterBenchmarks.testJdbcToArrow       avgt    5  5786.141 ±  68.501  us/op

After: 
JdbcAdapterBenchmarks.consumeBenchmark      avgt    5   554.501 ±  17.261  us/op
JdbcAdapterBenchmarks.consumeRowsBenchmark  avgt    5   558.829 ±  22.413  us/op
JdbcAdapterBenchmarks.testJdbcToArrow       avgt    5  5757.178 ± 197.539  us/op

File: java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/TimeConsumer.java [hunk 1]
```diff
@@ -42,56 +42,26 @@ public static TimeConsumer createConsumer(
...
+  static class NullableTimeConsumer extends BaseConsumer<TimeMilliVector> {
+
+    protected final Calendar calendar;
 
     /**
      * Instantiate a TimeConsumer.
      */
     public NullableTimeConsumer(TimeMilliVector vector, int index) {
-      super(vector, index);
+      this(vector, index, null);
```
Comment (emkornfield) [inline java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/TimeConsumer.java]: /*calendar=*/null for last parameter please.

File: java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
  * Consumer which consume array type values from {@link ResultSet}.
  * Write the data to {@link org.apache.arrow.vector.complex.ListVector}.
  */
-public abstract class ArrayConsumer implements JdbcConsumer<ListVector> {
+public abstract class ArrayConsumer extends  BaseConsumer<ListVector> {
```
Comment (emkornfield) [inline java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java]: spacing looks off

File: java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/TimeConsumer.java (hunk 1 above)
Comment (liyafan82) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/TimeConsumer.java]: Good point. Thanks.

File: java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java (hunk 2 above)
Comment (liyafan82) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java]: There is an extra space here. Thanks for the careful review. Maybe we need to enhace our style checking.

Comment (emkornfield) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java]: Yes, https://github.com/revelc/formatter-maven-plugin might be an option for this but we should discuss on the mailing list first (I would hope that we can configure it so not too many files need to change on the first pass).

Comment (liyafan82) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/ArrayConsumer.java]: Thanks for the suggested plugin. I agree with you that we should enhance the style checking in small steps.

**Review by xhochy — APPROVED** [review]
+1, LGTM, leaving for @emkornfield for a final review,.

**Pull request closed without merging**